components/HotWallet/plugins/TribecaGauges/EscrowData.tsx

```diff
@@ -0,0 +1,82 @@
+import { BN } from '@project-serum/anchor'
+import type { EscrowData } from '@tools/sdk/tribeca/programs'
+import { tryGetTokenMint } from '@utils/tokens'
+import BigNumber from 'bignumber.js'
+import { useCallback, useEffect, useState } from 'react'
+import useWalletStore from 'stores/useWalletStore'
+
+function formatDate(dateInSec: BN): string {
+  if (dateInSec.isZero()) {
+    return '-'
+  }
+
+  // mul by 1000 to get ms
+  return new Date(dateInSec.mul(new BN(1000)).toNumber()).toUTCString()
+}
+
+const EscrowDataBloc = ({ escrowData }: { escrowData?: EscrowData }) => {
+  const connection = useWalletStore((s) => s.connection)
+  const [uiAmount, setUiAmount] = useState<string>('-')
+
+  const loadUiAmount = useCallback(async (): Promise<string> => {
+    if (!escrowData) return '-'
+
+    const tokenInfo = await tryGetTokenMint(
+      connection.current,
+      escrowData.tokens
+    )
+
+    if (!tokenInfo) {
+      console.log(
+        'Cannot load information about token mint related to escrow data (tribeca gauges)',
+        escrowData.tokens
+      )
+      return '-'
+    }
+
+    return Number(
+      new BigNumber(escrowData.amount.toString())
+        .shiftedBy(-tokenInfo.account.decimals)
+        .toString()
+    ).toLocaleString()
```

Comment:
maybe you can refactor this into a utility function `formatAmountToLocalString(amount, decimals)` since you're doing the same at several places

Reply:
good idea :)